tests/test_api_typing.py

```diff
@@ -26,27 +30,32 @@
     Window,
 )
 import pytest
-from typing_extensions import (
-    assert_type,
-)
+from typing_extensions import assert_type
 
 from tests import (
     check,
     ensure_clean,
 )
 
-ResamplerGroupBy: TypeAlias = (
-    DatetimeIndexResamplerGroupby
-    | PeriodIndexResamplerGroupby
-    | TimedeltaIndexResamplerGroupby
-)
+if TYPE_CHECKING:
+    ResamplerGroupBy: TypeAlias = (
+        DatetimeIndexResamplerGroupby[Any]
+        | PeriodIndexResamplerGroupby[Any]
+        | TimedeltaIndexResamplerGroupby[Any]
+    )
+else:
+    ResamplerGroupBy: TypeAlias = (
+        DatetimeIndexResamplerGroupby
+        | PeriodIndexResamplerGroupby
+        | TimedeltaIndexResamplerGroupby
+    )
 
 
 def test_dataframegroupby() -> None:
     df = pd.DataFrame({"a": [1, 2, 3]})
     group = df.groupby("a")
 
-    def f1(gb: DataFrameGroupBy) -> None:
+    def f1(gb: "DataFrameGroupBy[Any, Any]") -> None:
```

Comment:
We need to be careful here. The generic forms are not documented, so we really should be testing just `DataFrameGroupBy` here in the tests because that would correspond to user code.

Maybe you can turn off that type checking for the `test` files in the CI.  

This would apply to other changes in this file as well